casper/src/main/resources/RevVault.rho

```diff
@@ -31,11 +31,11 @@ in {
       @MakeMint!(*mintCh) |
       for (mint <- mintCh) {
 
-        //FIXME the result of createGenesisVault is not protected, nor is the call to it. Both should be.
-        contract RevVault(@"createGenesisVault", @ownerPubKeyBytes, @initialAmount, ret) = {
-          new createGenesisVault in {
-            _getOrCreate!(ownerPubKeyBytes, *createGenesisVault, *ret) |
-            for (retCh <- createGenesisVault) {
+        //FIXME the result of findOrCreateGenesisVault is not protected, nor is the call to it. Both should be.
```

Comment:
Do we even need a "findOrCreateGenesisVault"?  I think we would be better to initialize the RevAddress Map with the proper address balances so they end up in the genesis block.

Reply:
The map doesn't store the balances. While hardcoding the genesis vaults creation along with their corresponding pubKeys could be an option, more questions remain:
1. how to secure access to those vaults?
2. how to put funds in them without providing a pre-filled purse to be used internally? (we probably could call _deposit directly from the revvault contract, but this needs some thought)

Long story short, this aspect hasn't yet been addressed. Do we need those sorted out for the demo?

Reply:
Oh I see I've just answered two of my questions :) The last one remains though :)

Reply:
I was sloppy with my words and didn't intend to imply balances are kept anywhere but the purse in the vault. But before I clarify, and repose a suggestion I have a question:

In your implementation is there a map storing **Vaults** (key being a PublicKey):
https://github.com/rchain/rchain/blob/dev/casper/src/main/resources/RevVault.rho#L30
And a tuple storing purses (uniquely keyed by the Rev Addresses):
https://github.com/rchain/rchain/blob/dev/casper/src/main/resources/RevVault.rho#L130

Reply:
I'm not sure I get the question. There is a pubKey -> vault, but there isn't another map. The second thing you're pointing to is a contract, registered according to the address scheme. We might leverage the `_deposit` contract by addressing it directly (not from within `transfer`) in some init code in the RevVault. What's your suggestion?